provider/adapters/src/fork/predicates.rs

```diff
@@ -28,7 +29,7 @@ pub trait ForkByErrorPredicate {
     /// [`DataKey`]: icu_provider::DataKey
     /// [`DataRequest`]: icu_provider::DataRequest
     /// [`IterableDynamicDataProvider`]: icu_provider::datagen::IterableDynamicDataProvider
-    fn predicate(key: DataKey, req: Option<DataRequest>, err: DataError) -> bool;
+    fn predicate(&self, key: DataKey, req: Option<DataRequest>, err: DataError) -> bool;
```

Comment:
A predicate's single method is usually called `test`. Otherwise you end up with things like `self.predicate.predicate(...)` which is weird.